### ISSUE TITLE
Can't update identities

### ISSUE THREAD

**Issue by hyw** [facetoe/zenpy#207]
I'm experiencing this error when trying to update a `phone_number` type identity for a given user.
```
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "/Users/yangwang/.pyenv/versions/3.6.2/lib/python3.6/site-packages/zenpy/lib/util.py", line 105, in inner
    return func(*new_args, **new_kwargs)
  File "/Users/yangwang/.pyenv/versions/3.6.2/lib/python3.6/site-packages/zenpy/lib/api.py", line 604, in update
    return UserIdentityRequest(self).put(self.endpoint.update, user, identity)
  File "/Users/yangwang/.pyenv/versions/3.6.2/lib/python3.6/site-packages/zenpy/lib/request.py", line 197, in put
    return self.api._put(url, payload=payload)
  File "/Users/yangwang/.pyenv/versions/3.6.2/lib/python3.6/site-packages/zenpy/lib/api.py", line 87, in _put
    response = self._call_api(self.session.put, url, json=self._serialize(payload), timeout=self.timeout)
  File "/Users/yangwang/.pyenv/versions/3.6.2/lib/python3.6/site-packages/zenpy/lib/api.py", line 132, in _call_api
    self._check_response(response)
  File "/Users/yangwang/.pyenv/versions/3.6.2/lib/python3.6/site-packages/zenpy/lib/api.py", line 271, in _check_response
    raise APIException(json.dumps(_json), response=response)
zenpy.lib.exception.APIException: {"error": {"title": "Invalid attribute", "message": "You passed an invalid value for the id attribute. Invalid parameter: id must be an integer"}}
```

The syntax I'm using to try and update an identity is:

```
zenpy_client.users.identities.update(
     user=u.id,
     identity=Identity(id=i.id, value='+19043005880'),
)
```

**Comment by facetoe** [comment]
Can you enable logging and post the output:

```
log = logging.getLogger()
log.setLevel(logging.DEBUG)
ch = logging.StreamHandler(sys.stdout)
formatter = logging.Formatter('%(levelname)s - %(message)s')
ch.setFormatter(formatter)
log.addHandler(ch)
logging.getLogger("requests").setLevel(logging.WARNING)
```

**Comment by hyw** [comment]
```

**Comment by facetoe** [comment]
Thanks, I've patched this bug.

**Comment by hyw** [comment]
@facetoe thank you!

**Comment by hyw** [comment]
@facetoe hey it seems that if I update a user's phone number, a new identity is added to the user's account. Could this be a bug due to the fix implemented for this?

For example, with this user.phone update:
```
>>> user = zenpy_client.users(id=27254634488)
>>> user.phone
'502-394-9400 x367'
>>> user.phone = '+15023949400x367'
>>> updated_user = zenpy_client.users.update(user)
>>> updated_user.phone
'+15023949400x367'
```
You see that the user's phone is updated accordingly, but, the Zendesk UI actually shows a second phone number added to the system (as opposed to the desired behavior of updating the original phone number):
<img width="288" alt="screen shot 2018-03-28 at 11 21 49 am" src="https://user-images.githubusercontent.com/1979879/38048455-430c353a-327a-11e8-879f-8466f2a4e66c.png">
An added oddity is that this newly added phone number doesn't appear when querying the `Identities` API and the older phone number is being returned:
```
>>> user_identities = list(zenpy_client.users.identities(id=27254634488))
>>> user_identities
[Identity(id=9686312208), Identity(id=9718761327)]
>>> for identity in user_identities:
...     print("Key: {}, Value: {}".format(identity.type, identity.value))
Key: email, Value: redacted@redacted.com
Key: phone_number, Value: 502-394-9400 x367
```

**Comment by facetoe** [comment]
Hmm I don't think this is a bug in Zenpy, just some surprising behavior in the Zendesk API. See https://developer.zendesk.com/rest_api/docs/core/users#phone-number. It looks like phone numbers are a bit of a special case.

I did a few tests with this and notice that each time the phone number is updated a new entry is shown in the UI, which does seem a bit strange. It might be worth reaching out to Zendesk to see if this is a bug on their end. I confirmed the same behavior is present when using curl:

```bash
curl https://mysubdomain.zendesk.com/api/v2/users/116514121092.json \
  -d '{"user": {"phone": "12345567999991"}}' \
  -H "Content-Type: application/json" -X PUT \
  -v -u 'email:password'
```
